.github/workflows/build.yml

```diff
@@ -122,12 +122,40 @@ jobs:
         run: cargo build --workspace --all-targets --all-features ${{ matrix.config.extraArgs }}
 
       - name: Cargo Test
+        if: ${{ fromJSON(matrix.config.os != 'ubuntu-latest') }}
         run: |
           make test-unit
           make test-e2e
         env:
           RUST_LOG: spin=trace
 
+      - name: Install cargo-tarpaulin binary crate (ubuntu)
+        uses: actions-rs/install@v0.1
+        if: ${{ fromJSON(matrix.config.os == 'ubuntu-latest') }}
+        with:
+          crate: cargo-tarpaulin
+          version: latest
+          use-tool-cache: true
+
+      - name: Cargo Tarpaulin (ubuntu)
+        if: ${{ fromJSON(matrix.config.os == 'ubuntu-latest') }}
+        run: cargo tarpaulin --follow-exec --skip-clean -t 6000 --out xml --features openssl/vendored,default,e2e-tests
+        env:
+          RUST_LOG: spin=trace
+
+      - name: Upload to codecov.io (ubuntu)
+        uses: codecov/codecov-action@v1.0.2
+        if: ${{ fromJSON(matrix.config.os == 'ubuntu-latest') && github.ref == 'refs/heads/main' }}
+        with:
+          token: ${{secrets.CODECOV_TOKEN}}
+
+      - name: Archive code coverage results (ubuntu)
+        uses: actions/upload-artifact@v1
```

Comment:
In the continued spirit of bumping to latest versions, can we bump this one as well?  Looks like v3.x is the current one...

Reply:
Updated it. Thanks.